src/libponyc/expr/reference.c

```diff
@@ -675,7 +675,7 @@ bool expr_reference(pass_opt_t* opt, ast_t** astp)
 
       ast_t* type = ast_type(def);
 
-      if(is_typecheck_error(type))
+      if(is_typecheck_error(type) && (errors_get_count(opt->check.errors) > 0))
```

Comment:
The reason I did this is because the approach of removing the line entirely made the error message for another badpony case a little worse: https://github.com/ponylang/ponyc/pull/1339/files#diff-6adc6b8a17f2bbfe51cbc0151f770ce5R263

However, I've updated the PR to use that approach - for now we can live with those consequences I think, and possibly try to improve that other error message in a separate PR.